parquet-tools/src/test/java/org/apache/parquet/tools/read/TestSimpleRecordConverter.java

```diff
@@ -0,0 +1,147 @@
+        switch(value.getName()) {
+          case INT32_FIELD:
+            Assert.assertEquals(32, value.getValue());
+            break;
+          case INT64_FIELD:
+            Assert.assertEquals(64L, value.getValue());
+            break;
+          case FLOAT_FIELD:
+            Assert.assertEquals(1.0f, value.getValue());
+            break;
+          case DOUBLE_FIELD:
+            Assert.assertEquals(2.0d, value.getValue());
+            break;
+          case BINARY_FIELD:
+            Assert.assertArrayEquals("foobar".getBytes(), (byte[])value.getValue());
+            break;
+          case FIXED_LEN_BYTE_ARRAY_FIELD:
+            Assert.assertArrayEquals(new byte[]{ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12 }, (byte[])value.getValue());
+            break;
+        }
+      }
+    }
+  }
+
+  @Before
+  public void setUp() throws IOException {
+    this.testFile = createTempFile();
+    MessageType schema = createSchema();
+    write(schema, testFile);
+  }
+
+  @After
+  public void tearDown() {
+    this.testFile.delete();
+  }
+
+  private MessageType createSchema() {
+    return new MessageType("schema",
+      new PrimitiveType(Type.Repetition.REQUIRED, PrimitiveType.PrimitiveTypeName.INT32, INT32_FIELD),
+      new PrimitiveType(Type.Repetition.REQUIRED, PrimitiveType.PrimitiveTypeName.INT64, INT64_FIELD),
+      new PrimitiveType(Type.Repetition.REQUIRED, PrimitiveType.PrimitiveTypeName.FLOAT, FLOAT_FIELD),
+      new PrimitiveType(Type.Repetition.REQUIRED, PrimitiveType.PrimitiveTypeName.DOUBLE, DOUBLE_FIELD),
+      new PrimitiveType(Type.Repetition.REQUIRED, PrimitiveType.PrimitiveTypeName.BINARY, BINARY_FIELD),
+      new PrimitiveType(Type.Repetition.REQUIRED,
+        PrimitiveType.PrimitiveTypeName.FIXED_LEN_BYTE_ARRAY, 12, FIXED_LEN_BYTE_ARRAY_FIELD)
+    );
+  }
+
+  private void write(MessageType schema, File f) throws IOException {
+    Path fsPpath = new Path(f.getPath());
+    Configuration conf = new Configuration();
+    SimpleGroupFactory fact = new SimpleGroupFactory(schema);
+    GroupWriteSupport.setSchema(schema, conf);
+
+    ParquetWriter<Group> writer = new ParquetWriter<Group>(
+      fsPpath,
+      new GroupWriteSupport(),
+      CompressionCodecName.UNCOMPRESSED,
+      1024,
+      1024,
+      512,
+      true,
+      false,
+      ParquetProperties.WriterVersion.PARQUET_2_0,
+      conf);
+    try {
+      writer.write(fact.newGroup()
+       .append(INT32_FIELD, 32)
+       .append(INT64_FIELD, 64L)
+       .append(FLOAT_FIELD, 1.0f)
+       .append(DOUBLE_FIELD, 2.0d)
+       .append(BINARY_FIELD, Binary.fromString("foobar"))
+       .append(FIXED_LEN_BYTE_ARRAY_FIELD,
+         Binary.fromConstantByteArray(new byte[]{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12 })));
+    } finally {
+      writer.close();
+    }
+  }
+
+  private File createTempFile() throws IOException {
+    File tmp = File.createTempFile(getClass().getSimpleName(), ".tmp");
```

Comment:
I would replace this one with a junit rule: https://garygregory.wordpress.com/2010/01/20/junit-tip-use-rules-to-manage-temporary-files-and-folders/

Reply:
@Fokko I did not know about Temporary Folder. Thanks !!